tests/framework/web/ViewTest.php

```diff
@@ -137,4 +138,93 @@ private function getCSRFTokenValue($html)
 
         return $matches[1];
     }
+
+    private function setUpAliases() {
+        \Yii::setAlias('@web', '/');
+        \Yii::setAlias('@webroot', '@yiiunit/data/web');
+        \Yii::setAlias('@testAssetsPath', '@webroot/assets');
+        \Yii::setAlias('@testAssetsUrl', '@web/assets');
+        \Yii::setAlias('@testSourcePath', '@webroot/assetSources');
+    }
+
+    public function testAppendTimestampForRegisterJsFile()
+    {
+        $this->mockWebApplication([
+            'components' => [
+                'request' => [
+                    'scriptFile' => __DIR__ . '/baseUrl/index.php',
+                    'scriptUrl' => '/baseUrl/index.php',
+                ],
+            ],
+        ]);
+
+        $this->setUpAliases();
+
+        $pattern = '/js\/jquery\.js\?v\=/';
```

Comment:
Can you change this to
```suggestion
        $pattern = '/js\/jquery\.js\?v\=\d+"/';
```

to make sure that timestamp is appended only once?